bee-api/bee-rest-api/src/endpoints/routes/health.rs

```diff
@@ -1,51 +1,28 @@
-// Copyright 2020-2021 IOTA Stiftung
+// Copyright 2020-2022 IOTA Stiftung
 // SPDX-License-Identifier: Apache-2.0
 
 use std::{
     convert::Infallible,
-    net::IpAddr,
-    time::{SystemTime, UNIX_EPOCH},
+    time::{Duration, SystemTime, UNIX_EPOCH},
 };
 
+use axum::{extract::Extension, http::StatusCode, response::IntoResponse, routing::get, Router};
 use bee_protocol::workers::PeerManager;
-use bee_runtime::resource::ResourceHandle;
 use bee_tangle::Tangle;
-use warp::{filters::BoxedFilter, http::StatusCode, Filter, Reply};
 
-use crate::endpoints::{
-    config::ROUTE_HEALTH,
-    filters::{with_peer_manager, with_tangle},
-    permission::has_permission,
-    storage::StorageBackend,
-};
+use crate::endpoints::{storage::StorageBackend, ApiArgsFullNode};
 
 const HEALTH_CONFIRMED_THRESHOLD: u32 = 2; // in milestones
-const HEALTH_MILESTONE_AGE_MAX: u64 = 5 * 60; // in seconds
-
-fn path() -> impl Filter<Extract = (), Error = warp::Rejection> + Clone {
-    warp::path("health").and(warp::path::end())
-}
+const HEALTH_MILESTONE_AGE_MAX: Duration = Duration::from_secs(5 * 60);
 
-pub(crate) fn filter<B: StorageBackend>(
-    public_routes: Box<[String]>,
-    allowed_ips: Box<[IpAddr]>,
-    tangle: ResourceHandle<Tangle<B>>,
-    peer_manager: ResourceHandle<PeerManager>,
-) -> BoxedFilter<(impl Reply,)> {
-    self::path()
-        .and(warp::get())
-        .and(has_permission(ROUTE_HEALTH, public_routes, allowed_ips))
-        .and(with_tangle(tangle))
-        .and(with_peer_manager(peer_manager))
-        .and_then(|tangle, peer_manager| async move { health(tangle, peer_manager) })
-        .boxed()
+pub(crate) fn filter<B: StorageBackend>() -> Router {
+    Router::new().route("/health", get(health::<B>))
 }
 
-pub(crate) fn health<B: StorageBackend>(
-    tangle: ResourceHandle<Tangle<B>>,
-    peer_manager: ResourceHandle<PeerManager>,
-) -> Result<impl Reply, Infallible> {
-    if is_healthy(&tangle, &peer_manager) {
+pub(crate) async fn health<B: StorageBackend>(
```

Comment:
For this an all similar function in all other module, make them private if they don't actually need to be pub crate

Reply:
done